src/main/java/Storage.java

```diff
@@ -0,0 +1,130 @@
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.time.LocalDate;
+import java.util.List;
+
+/**
+ * deals with loading tasks from the file and saving tasks in the file
+ */
+public final class Storage {
+    private static final String DB_DIR_NAME = "data";
+    private static final String DB_FILE_NAME = "duke.txt";
+    private static Path dbPath = null;
+
+    private TaskList toDoLst;
+
+    public Storage() throws FileNotFoundException, IOException {
+        try {
+            checkDbExists();
+        } catch (FileNotFoundException e) {
+            throw e;
+        }
+
+        toDoLst = new TaskList();
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(dbPath.toFile()));
+            String line = br.readLine();
+
+            while (line != null) {
+                String[] lineData = line.split("\\|");
+
+                Task task = new Task(lineData[0], lineData[2], lineData.length == 3 ? null : LocalDate.parse(lineData[3]), Boolean.valueOf(lineData[1]));
+
+                addToDoItem(task);
+
+                line = br.readLine();
+            }
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Checks whether the text file (for storage) exists
+     *
+     * @throws FileNotFoundException
+     */
+    public static void checkDbExists() throws FileNotFoundException {
+        Path curDirPath = Paths.get("");
+        Path dbDirPath = curDirPath.resolve(DB_DIR_NAME);
+
+        if (!Files.exists(dbDirPath)) {
+            throw new FileNotFoundException(String.format("Data folder does not exist! Please create a directory named %s", DB_DIR_NAME));
+        }
+
+        dbPath = dbDirPath.resolve(DB_FILE_NAME);
+
+        if (!Files.exists(dbPath)) {
+            throw new FileNotFoundException(String.format("Db does not exist! Please create a text file named %s", DB_FILE_NAME));
+        }
+    }
+
+    /**
+     * Get to-do list.
```

Comment:
The javadoc comment would be great if if could start with a simple present tense verb like 'Gets'.